src/client.js

```diff
@@ -45,8 +45,8 @@ class Client {
     //         // append the debug log to a #debug div
     //         $("#debug").append(str + "\n");
     //     };
-    debug(...args) {
-        if (this.hasDebug) console.log(...args);
+    debug(message, frame) {
```

Comment:
If you are not going to use `frame` inside the function, passing it as a parameter is redundant.

Reply:
true, I will remove the param

Reply:
Removing the param raises another question. The whole PR ends up moving ` let frame = new Frame(command, headers, body);` from `marshall` to `_tramsit` with no obvious gain as `frame` will not be used as a second param to `debug` function. The PR looks kinda pointless. I wonder if there is something else I don't see.

Reply:
Yes, there is when overriding the `client.debug` then you have access to `command`, `headers` and `body` separately and can log what you need. We are for example truncation `Authorization` header:

```
client.debug = (text, frame) => {
    if (frame.headers && frame.headers.Authorization) {
      const newHeaders = evolve({
        Authorization: truncate
      }, frame.headers)
      log.debug(_frameToString(frame.command, newHeaders, frame.body))
    }
  }
```

It's really much easier to read the logs now as our JWT was ocupying 3 console lines. We may soon strip some content of the `body` or completely remove some headers.

Reply:
And why is the current signature not suiting you? I mean from what I know `(text, frame)` complies with `(...args)`. Declaring 

```
client.debug = (...args) => {
    if (arg[1].headers && arg[1].headers.Authorization) {
      const newHeaders = evolve({
        Authorization: truncate
      }, arg[1].headers)
      log.debug(_frameToString(arg[1].command, newHeaders, arg[1].body))
    }
  }
```
and calling `debug(text, frame)` should work the same.

Reply:
You're kinda right but generali when there is a logging function and has `...args` it's supposed that it will directly log all the varargs. 

A better signature would be `client.debug (frame, ...args) => ...` but obviously it's not backward compatible. 

So when we know that the second arg will always be a frame why not calling it `frame`? And clockwise, the 1st param will always be a message...

Reply:
It's supposed to log whatever arguments you are passing into the function. And having the parameters signature as `...args` gives all the power to the handler function. Just handle it correctly.

From my POV if I when @JSteunou I wouldn't accept the PR for the following reasons:
1. Changing the current signature to strictly two arguments (text, frame)  will limit the logging capability.
2. Introduces breaking change to the base code resulting in the need to bump the major version.

And to avoid any misunderstandings, I'm just expressing my opinion here. I'm not affiliated with the repo and my opinion does not reflect the opinion of the repo owner.

Reply:
No problem, thank you for your opinion.

To my defense, the logs as they are used now are using only 1 message. I could have mistaken but I went through all of the usages of `.debug()`. People who overridden `debug` maybe in trouble, that's true. They're mostly destructuring the `args`. Maybe creating another method, e.g. `debugWithFrame(frame, ...args)`, would be helpful.

Reply:
Just out of curiosity are you using vanilla `javascript`?

Reply:
I am using ES6 but it can be rewritten to vanilla very easily. Maybe the [ramda](http://ramdajs.com/) library is confusing you?

Reply:
I'm just asking cause I realized that we are not providing typescript `types` for the `debug` function and this will definitely cause problems for those who want to override the function.

Reply:
I've never used Typescript but it's another obstacle. What do you say about using the other function `debugWithFrame(frame, ...args)`. I definitely think it's necessary for us to have a finer way to pick what we would include in our logs than parsing a string.

Reply:
I'm not familiar with `Ramda` (man there are so many libraries out there), but I stick to `Zakas` quote that `named arguments in ECMAScript are a convenience and not a necessity`.

Reply:
Nevermind, why not. I'll wait with further changes until we agree on them. @JSteunou @JimiC